### DISCUSSION TITLE
Close discussions

### DISCUSSION THREAD

**Discussion by TrueBrain** [community/community#3097] [Discussions] [answered]
As it goes with any discussion, there is a natural point a discussion comes to an end.

Sadly, currently, I cannot find a way to close a discussion .. only to lock the conversation.

The reason we would like this:

Currently we are using Discussions to talk over feature requests, as it has a more natural format for that (being able to reply to a comment is really helpful here). But as time goes on, either the feature request gets implemented or is rejected. In either case, we would like to mark the Discussion as such, without being all evil and locking anyone from continue the conversation if there is a need. Very similar to issues and PRs.

Now we notice people are unaware a feature has either been rejected or implemented, and keep contributing to something already decided. This requires us to change the original post to note this, but it is a very annoying workflow. Especially as there is also no way to label discussions.

Easier would be if there was a category-type that would lend itself for this workflow, where we can just mark a Discussion as "closed" or "resolved" or similar. Not so much as a QA, as there is no right answer.

Would this be possible? Or is it already, and am I just overlooking this?

Thank you!

**Comment by gabro** [comment]
Same exact use case here: I'm using discussions for feature requests and it would be very useful to mark them as done/rejected. 

Right now there's no way of looking at the feature requests list and tell which one are pending and which one are completed.

Alternatively to closing discussions, it would be interesting to be able to convert them to issues (which would enable to close them via a PR, for example)

**Reply by ouuan** [reply]
I don't think Discussions is a place for feature requests. Actually I use "an issue is what can be closed/resolved by a pull request" as the rule to decide when to open an issue and when to open a discussion. However, you certainly can have a different rule.

However, I still want discussions to be closable. A discussion should be closed when the conversation is resolved, while an issue should be closed by a pull request/commit.

**Reply by gabro** [reply]
I mentioned feature requests because "Ideas for new features" is literally one the default categories when you enable Discussions on a project:

![image](https://user-images.githubusercontent.com/691940/105464960-9fbee300-5c92-11eb-9507-f507da66bf22.png)

If feature requests don't belong to Discussions, then I would propose to remove that default option because it's quite crucial for feature requests to be addressable in some way (mark them as done, rejected, or in progress, for instance)

**Reply by ouuan** [reply]
AFAIK, there's no public-accepted convention of the difference between issues and discussions. But I believe it doesn't make sense to use different systems for feature requests and bug reports.

On the first day Discussions was in public beta, another maintainer of [CP Editor](https://github.com/cpeditor/cpeditor/) converted several feature requests into discussions without my acknowledgment, so now I'm eagerly waiting for #2861. So I'd agree that it shouldn't be encouraged to use Discussions for feature requests in the UI of GitHub.

**Reply by gabro** [reply]
In my experience feature requests tend to be very noisy (since anyone can propose one and then they stay open for a long time) and I've always liked managing them separately than issue trackers.

For instance, in [Metals](https://github.com/scalameta/metals) we use a [separate repo](https://github.com/scalameta/metals-feature-requests ) for tracking feature requests and we use a GitHub action to generate a summary table sorted by 👍  reactions in a [pinned issue](https://github.com/scalameta/metals/issues/707) in the main repo.

This way we have a manageable issue tracker focused on bugs and actionable features, while all the ideas can float around in the other repo.

I think Discussions with upvoting would be great for this use-case, if only we could then either close them or turn them into issues when we start working on them.

**Reply by ouuan** [reply]
> actionable features

That's what I think "can be closed by a PR". I agree that an immature idea should be in the Discussions.

**Reply by sebersole** [reply]
> AFAIK, there's no public-accepted convention of the difference between issues and discussions. But I believe it doesn't make sense to use different systems for feature requests and bug reports.

I'll disagree here.  We use Jira for issue tracking (including feature requests).  However, Discussions is a much better format for, well, discussions about features as well as other things that do not fit into issue tracking.  Granted, eventually feature discussions get to a "resolution" at which point there would be a Jira created with a link to the discussion.

Jira is simply not a good tool for this

**Comment by evi-liu** [comment]
Thanks for the feedback! We are tracking this request on our end and using this thread to gauge customer impact via upvotes and reactions.

**Reply by abravalheri** [reply]
Hi @evi-liu, I suppose this feature is a generalisation of the "Answered" thing for questions, but that work for all the categories...

**Reply by kaitlin-beegle** [reply]
Hi @evi-liu, any update on the feasibility of this feature being incorporated? It has quite a few upvotes, and we at Filecoin Foundation and Protocol Labs would also benefit from this enhancement.

**Comment by jay** [comment]
I would also like this. It cannot be determined from the [discussions listing](https://github.com/curl/curl/discussions) whether a "general discussion" has been resolved. For Q&A there's a green check mark but for general discussion the options are lock or delete. I don't want to do either, I just want some way to mark it that the discussion is resolved. If a general discussion can't be answered, it can be resolved /ended with a green check mark. Right now we have two general discussions, [one has resolved itself](https://github.com/curl/curl/discussions/6611), and [one has ended](https://github.com/curl/curl/discussions/6466), but no way to signal they've ended.

**Comment by jsmorabito** [comment]
Another use case from my team at Athens Research https://github.com/athensresearch/athens/blob/master/README.md

We are an active and growing community around an open source product and are looking for ways to make feature requests more tractable. The voting system in the discussions section is great because it allows us to see which features are the most requested (or will at least when we get more momentum using it for feature requests). I'm already anticipating the challenges mentioned by posters above. I'm imagining that where you are able to select the discussion format (currently the options are Q&A and open discussion) you might have a feature request format that allows discussions to be marked as closed or resolved. Likewise I could see a discussion format option with custom parameters that might allow for other use cases as well. Looking forward to a solution. If I can be of any more help please let me know! Thanks!

**Comment by bartr** [comment]
+1 for this feature. We're using Discussions for our Triage and it's working well for us.

As a temporary workaround, we created an `Archive` category and just change the category. The ability to `close`  or `archive` (like you do notes on a Project) would be perfect for us.

**Reply by riverar** [reply]
Great workaround, we're going to try that as well for now.

Downsides:
* Makes it appear as if the user originated the discussion in that category
* Category changes are not tracked

**Reply by tayloramurphy** [reply]
Agreed on the downsides. I think we may "solve" this by updating the description with the changes.

**Comment by senorflor** [comment]
Another workaround: we hit upon the idea of locking obsolete discussions and then filtering the top view.

`is:unlocked` is then sufficient to show us then only the current discussions.

Proposed for prioritization in https://github.com/github/feedback/discussions/3568

**Comment by adityaruplaha** [comment]
I also have a similar usecase, features have already been implemented but the discussion didn't reflect that. A way to a close/archive discussions would be much appreciated.

**Comment by Relequestual** [comment]
We at JSON Schema are trialing Discussions as a non-ephemeral (*ahem* free Slack) place to make suggestions about how to operate as an open source community, project vision, and our direction. We would love to be able to mark discussions as resolved without locking them, and have some sort of default filter.

Resolved / archived seems like the right thing here. Discussions that result in actions will be tracked by Issues if a consensus of action is reached, but we got to get there first. My feeling is we want to rasie an Issue when there's work to be done.

**Comment by azinit** [comment]
still actual!

**Comment by moving-bits** [comment]
also see https://github.com/github/feedback/discussions/4026

**Comment by SmartieCodes** [comment]
I would love to see an archive option for a discussion, something that works like issues being closed. You can go to the closed/archived tab to see them if needed, but only active discussions are shown by default.

**Comment by michael-hawker** [comment]
Yeah, I like the idea of having a 'close' state for discussions. If someone opens a discussion that's off-topic or in the wrong-place, there's not really a way to 'close' it and remove it from the top of the list of discussions.

You don't always want to delete these threads though as it may help folks searching with similar questions to get context or pointers to where to go for answers in the future.

**Comment by princestha** [comment]
I will upvote this feature. I would love for the ability to close discussions and archive them without deleting them. Once closed or archived, they shouldn't show up in the new or top. But can still be searched and possibly a selection called show archive to show the archived post to those who need it.

**Comment by mmartinbarr** [comment]
+1 to this. We need a way to visually filter out closed/locked discussions at the UI.

We have a huge number of discussions that were created under several categories (General, Ideas, even custom categories) and locking them does not mark them or label them anyhow at the "View all" UI, neither there is a filter for Locked/Unlocked discussions.

The workaround that we are thinking of doing in the meanwhile is just to create every new discussion under the Q&A category (so that at least we can mark them as "Answered") and change the category of the discussions already locked to Q&A... but that should be a workaround.

**Comment by kylefox** [comment]
Would definitely love to have "Close discussion" functionality that hides the discussion from the list.

I'd like our team to start using GitHub Discussions to talk through various technical/product decisions, and then close once a decision has been reached and/or the topic is no longer relevant.

For now I've been tagging these discussions as "Resolved", but it's definitely not ideal because they still show up alongside all the active discussions.

**Comment by canteversleep** [comment]
An organization I am a part of uses discussions for policies as part of a roadmap repo and would benefit a lot from this.

**Comment by cpholguera** [comment]
Close or Resolve discussion would be amazing so that they do not appear in the main list anymore. Right now it's very unintuitive and unclear. I loose a lot of time opening discussions that I think need an answer but they're already finished.

Maybe also add a button to see closed/resolved/archived discussions.

**Comment by alexellis** [comment]
Just had the first discussion on a private sponsors-only repo which was a bit of a distracting / off-topic question. It's valid but I don't want it polluting the top of the discussions list. Archiving it would be ideal.

I tried the "creating another category" workaround suggested here but it doesn't remove it from the front page of the "all discussions" section which is the default view.

**Comment by developerasun** [comment]
Yeah it would be very helpful if we have this feature in discussion. I like the way discussion works as newsfeed(latest one becomes the first line) but would be even more helpful with open/close/reject status feature other than category.

**Comment by ghost** [comment]
Hi all! 
1. I would like to help GitHub and this idea you guys are thinking about. reference: https://github.com/github/feedback/discussions/16739 
2. In short my idea is to take time to delete the discussions. Thus, newer discussions are started and old discussions are automatically deleted. 
3. So... software like Discourse has this feature, it would be interesting to close discussions or delete discussions as time goes by.
4. +1 for this feature.

**Comment by swilgosz** [comment]
I'm using discussions to collect ideas to my screencast episodes from community. When I address a suggested topic, I'd love to close discussion/archive it. I don't want to fully delete any threads though. +1 for this feature.

**Comment by hiragram** [comment]
+10000

**Comment by Jamelle-Boose** [comment]
The possibility of archiving a discussion would allow me to completely migrate from Notion. Heck, even the ability to set a default filter would be useful. However, I have to manually filter every time, which becomes tedious.

**Comment by rmathew1992** [comment]
Hi! I’m a staff member at GitHub, and I’ve begun looking into this request. I had a few follow-up questions that I was hoping y’all could help me with. I know parts of these questions are covered in earlier threads, so apologies for any repetition.

* Is the problem that people are looking to solve:
  * A desire to clean up the discussions list page to make it more useful?
  * Prevent spammy comments on particular discussions?
  * Something else? Some combination of these things?
* Is the current ‘lock a discussion' behavior different than how people would wish 'closing a discussion' to work? 
  * If so what is the difference?
  * Would defaulting the discussions list page to filter for `is: unlocked` accomplish the same thing as ‘closing a discussion

Thanks for all of your feedback so far!

**Reply by swilgosz** [reply]
For me if the default view would filter out all locked discussions, it would be enough. My usecase:

I use github discussions to collect topic ideas from my audience for my future blog posts. Once I adress such Idea I don't want it to be seen by default anymore.

At the same time I want to have history untouched so discussions are available for browsing, but not active for participating.

The key feature I need is to get them out of sight.

**Reply by shotarok** [reply]
Yes, the desire to clean up the discussions is my use case. We've been using the discussion to gather feature requests for our internal platform. I want to exclude completed feature requests by default somehow. And the default filter `is: unlocked` would also work for our use case.

**Reply by cpholguera** [reply]
Hi @rmathew1992, the problem now is that you loose a lot of time checking discussions that you think need an answer but they're already finished.


✅ Setting the default filter to `is: unlocked` should work fine.

 
Thank you very much!

**Reply by ouuan** [reply]
I would like to get "closed" discussions out of sight without locking them, because I often find it annoying to be unable to comment under a locked issue. Sometimes further discussions are needed, and I prefer locking only when there is a reason for it (e.g. for some reason, spam is likely to happen or is already happening).

**Reply by tayloramurphy** [reply]
> Is the problem that people are looking to solve

We use discussions for internal team conversations to make decisions and external open source conversations. The problems I'm trying to solve are:

* cleanup the listing page so only open and active discussions are shown
* prevent any more comments, even from members of the org (maintainers I suppose) from further discussion. locking has no real use for internal discussions.

That said, having a `is: unlocked` filter would be a step in the right direction.

**Reply by michael-hawker** [reply]
Locking is really strict measure that really prevents further engagement down the line if new or related issues arise. I rarely lock anything unless it'd be off-topic or spam.

Marking things as an answer and closing is really about saying this discussion has concluded and isn't needed to be bubbled up to others in the list by default (they should still be searchable).

**Reply by hoffa** [reply]
Echoing comments above, for us the goal is to have resolved open-ended conversations (typically feature requests) out of sight by default, without outright locking them (otherwise there's no ways to further comment without workarounds such as creating a new discussion).

**Reply by josephj11** [reply]
We also want this feature for AutoKey. Locking and deleting are usually not good options. Old information is often still valuable and, sometimes, further comments are valuable. We do need to get these old/finished discussions out of view.

**Comment by ghost** [comment]
Hi all. It would be interesting to archive discussion in Github discussions - instead of just ending a discussion. I agree with this feature, but... I would just include the idea of ​​archiving... as the topic could be reopened if requested.

**Comment by marikaner** [comment]
We just started discussing architecture and API decisions in GH discussions. Once a decision is made we would like to clean the discussions up. For me it is not relevant whether it becomes read-only or not. I merely need a visual clue whether this is still an active discussion or not.

**Comment by josephj11** [comment]
One thing that I haven't seen mentioned elsewhere is that discussions share the same numbering sequence as issues. If a bunch are added, subsequent issues get much higher numbers.

While this is largely a cosmetic issue, it would be nice if discussions were numbered separately somehow, but could still have the very nice feature of short bang (#) links in other issues/discussions.

Overall, having a way to close and/or archive discussions would be quite helpful.

**Comment by Elliria** [comment]
It would be a good idea to provide this feature.

**Comment by sebersole** [comment]
PSA : A vote is more than enough folks and won't continually ping everyone who has been watching this discussion for a long time

**Comment by ghost** [comment]
+ 1

**Comment by rib** [comment]
Related to this; I also wish there was a way to effectively close a discussion by converting it into an issue in situations where someone opens a discussion that the maintainers would rather treat as an issue.

This would act a bit like "Transfer this discussion" and "Create issue from discussion" combined.

Currently you can use "Create issue from discussion" and then delete discussion but you don't get a good paper trail of what your doing.

If you had a way to transfer a discussion to an issue then the original (deleted) discussion could simply redirect to the new issue and the new issue could have an annotation similar to what you get when transferring issues between repositories.

**Comment by abhinavvadrevu** [comment, marked as answer]
Hi all, we are really excited to introduce the ability to close Discussions! Please read more about the feature here: https://github.blog/changelog/2023-03-14-adding-the-ability-to-close-github-discussions/

**Reply by classabbyamp** [reply]
Thanks! Is it possible to use the `closes #n` or `fixes #n` keywords to close discussions on push/merge?

**Reply by ouuan** [reply]
> Thanks! Is it possible to use the `closes #n` or `fixes #n` keywords to close discussions on push/merge?

IMO, when you need this, it should be an issue instead of a discussion. For example, you may briefly discuss what to do in a discussion first. When the discussion is mature enough, close the discussion and open an issue with the details. Then use a PR to close the issue.

**Reply by classabbyamp** [reply]
My intended usage for this is ~feature requests (requesting a package be added to a distro). having hundreds of requests overwhelms the issue tracker and hides the actual bugs, so I don't think that'd work for me

**Reply by jay** [reply]
When I search 'is:open test' the text stays in the query box but when I search 'is:opened test' the 'is:opened' part disappears. I mention this since your announcement suggests use the latter. Also is there a way to mass close many discussions at once?

**Reply by kmk3** [reply]
> When I search 'is:open test' the text stays in the query box but when I
> search 'is:opened test' the 'is:opened' part disappears. I mention this since
> your announcement suggests use the latter.

I think it's a typo in the announcement.

`is:open` works for both issues and discussions.

Searching discussions with any invalid filters (such as `is:aaa`) makes them
disappear.

**Reply by TPS** [reply]
@TrueBrain Close FTW?

**Comment by markseuffert** [comment]
Is there a repository setting to disable or restrict this new feature to maintainers?

I understand that some want the ability to close/lock/delete discussions.  We're using discussions mainly for Q&A. A space where our community can ask and answer questions. I like to avoid users accidentally closing a question instead of marking it as answered. The work flow that I like to suggest for Q&A is as following: The primary way of "closing a question" is [marking it as answered](https://docs.github.com/en/discussions/managing-discussions-for-your-community/moderating-discussions#marking-a-comment-as-an-answer), if for some reason that's not possible, [a maintainer can close/lock/delete a discussion in the sidebar](https://docs.github.com/en/discussions/managing-discussions-for-your-community/managing-discussions). At the moment, closing a discussion is not restricted to maintainers. Let me know what you think.

**Reply by TPS** [reply]
Isn't Wiki better suited to this, once the question has been definitively answered?  So promoting a discussion to wiki might be the route.

**Comment by jamesgeorgewilliams** [comment]
It would be nice if you are able to disable this new feature, or limit closing to only maintainers/contributors of a repository.

If using discussions as a Q&A, `is:open` is the default filter. 

That means if a user searches discussions for their problem, and is unfamiliar with the UI, they might not find the solution as a `closed` discussion is not visible by default. This can lead to them creating a new discussion even though it has been answered already.

Limiting closing to maintainers/contributors, or a default filter of `All` discussions would be preferable for some cases.

**Reply by jamesgeorgewilliams** [reply]
Just to add to the point above from @markseuffert , we also have users who are satisfied with an answer, but closing the discussion instead of marking it as answered.

The original purpose of discussions was to:

> GitHub Discussions is a collaborative communication forum for the community around an open source or internal project. 

This new feature renders the `mark as answer` functionality of github, and the ` is:answered ` filter a little useless, as now the community may not find the answers they are searching for.

While this feature is great for teams discussing internal feature requests etc, the implementation could be better for a community Q&A.

**Comment by caco3** [comment]
I think it is great to be able to close discussions.

As was before, if no reply was right to be marked as "answered", one still had to pick one to kind of close a discussion (because it helps to see live discussions).

Now my only request would be to close discussions through actions.
Eg. I would like to close a discussion, if no reply gets marked as answer within a certain time. Obviously I would send a reminder beforehand. This would adapt the auto close behavior some actions provide for the issues.
